elastalert/alerters/opsgenie.py

```diff
@@ -76,17 +78,23 @@ def alert(self, matches):
         if self.teams:
             post['teams'] = [{'name': r, 'type': 'team'} for r in self.teams]
         post['description'] = body
-        post['source'] = 'ElastAlert'
+        if self.entity:
+            post['entity'] = self.entity.format(**matches[0])
+        if self.source:
+            post['source'] = self.source.format(**matches[0])
 
         for i, tag in enumerate(self.tags):
             self.tags[i] = tag.format(**matches[0])
         post['tags'] = self.tags
 
-        if self.priority and self.priority not in ('P1', 'P2', 'P3', 'P4', 'P5'):
+        priority = self.priority
+        if priority:
+            priority = priority.format(**matches[0])
+        if priority and priority not in ('P1', 'P2', 'P3', 'P4', 'P5'):
```

Comment:
nit:  Could this block be indented as well?  The second `priority` check won't be successful if the first one fails